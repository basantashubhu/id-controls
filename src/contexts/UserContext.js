import {createContext, useContext, useEffect, useState} from "react";
import {auth} from '../firebase'

export const UserContext = createContext(null);

export function useAuth() {
    return useContext(UserContext)
}

export function AuthProvider({children}) {
    const [currentUser, setCurrentUser] = useState()
    const [loading, setLoading] = useState(false)
    const signUp = (email, password) => {
        return auth.createUserWithEmailAndPassword(email, password);
    }
    const logout = () => {
        return auth.signOut()
    }
    const login = (email, password) => {
        return auth.signInWithEmailAndPassword(email, password)
    }
    const resetPassword = (email) => {
        return auth.sendPasswordResetEmail(email)
    }
    useEffect(() => {
        return auth.onAuthStateChanged(user => {
            setCurrentUser(user);
            setLoading(false);
        })
    }, []);
    const value = {
        currentUser,
        signUp,
        logout,
        login,
        resetPassword
    };
    return <UserContext.Provider value={value}>
        {!loading && children}
    </UserContext.Provider>
}