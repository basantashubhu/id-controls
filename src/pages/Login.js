import {useState} from "react";
import {useAuth} from "../contexts/UserContext";
import {Link, Redirect, useHistory} from "react-router-dom";

const Login = () => {
    const history = useHistory()
    const {login, currentUser} = useAuth()
    const [credential, setCredential] = useState({email: '', password: ''})
    const [error, setError] = useState('')
    const [loading, setLoading] = useState(false)
    const handleChange = (e) => {
        setCredential({...credential, [e.target.name]: e.target.value})
    }
    const handleLogin = async (e) => {
        e.preventDefault();
        try {
            setError('');
            setLoading(true);
            await login(credential.email, credential.password);
            history.push('/')
        } catch (e) {
            setError('Failed to log in : ' + e.message);
        } finally {
            setLoading(false)
        }
    }
    return <>
        <div className="container">
            <div className="d-flex justify-content-center">
                <div className="w-100" style={{maxWidth: '400px'}}>
                    <div className="card">
                        <div className="card-body">
                            <h2 className="text-center mb-4">Log in</h2>
                            {error && <div className="alert alert-danger">{error}</div>}
                            <form onSubmit={handleLogin}>
                                <div className="form-group">
                                    <label>Email</label>
                                    <input type="email" name="email" className="form-control" value={credential.email}
                                           onChange={handleChange} required/>
                                </div>
                                <div className="form-group">
                                    <label>Password</label>
                                    <input type="password" name="password" className="form-control"
                                           value={credential.password} onChange={handleChange} required/>
                                </div>
                                <div className="form-group">
                                    <button type="submit" className="btn btn-primary w-100" disabled={loading}>Login</button>
                                </div>
                            </form>
                            <div className="text-center mt-3">
                                <Link to={'forgot-password'}>Forgot Password?</Link>
                            </div>
                        </div>
                    </div>
                    <p className="text-center mt-3">Need an account? <Link to={'/sign-up'}>Sign Up</Link></p>
                </div>
            </div>
        </div>
    </>
}

export default Login