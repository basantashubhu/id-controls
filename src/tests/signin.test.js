import React from 'react'
import {shallow, configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Login from '../pages/Login'
import * as UserContext from '../contexts/UserContext'
configure({adapter :  new Adapter()})
describe('Test case for login', () => {
    let wrapper, useAuth;
    beforeAll(() => {
        useAuth = jest.spyOn(UserContext, 'useAuth')
    });
    beforeEach(() => {
        useAuth.mockClear();
        useAuth.mockReturnValue(() => ({
            login : () => 1
        }))
    })
    it('email check', () => {
        wrapper = shallow(<Login/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('[name="email"]')).toHaveLength(1);
    })
    it('password check', () => {
        wrapper = shallow(<Login/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('[name="password"]')).toHaveLength(1);
    })
})
