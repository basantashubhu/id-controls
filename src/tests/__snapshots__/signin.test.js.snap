// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for login email check 1`] = `ShallowWrapper {}`;

exports[`Test case for login password check 1`] = `ShallowWrapper {}`;
